Accept sorting and listing options for all subreddits functions

aggrolite opened the pull request:
Allow the user to provide sorting and listing options
for all subreddit functions, most notably Frontpage() and
DefaultFrontpage(). This change also removes SortedSubmissions(),
which was made obsolete now that subreddit functions consistently
accept the same options.

---

**MAJOR CHANGES. FEEDBACK WANTED.**

I was playing around with functions `DefaultFrontpage()` and `Frontpage()` when I noticed that I could not do usual subreddit actions like sorting and limiting.

After looking over the function `SortedSubmissions()`, I realized it might be better to provide the same arguments for all subreddit-type functions. Except, functions `DefaultFrontpage()` and `Frontpage()` would not need a `subreddit` argument.

I think this makes the library more consistent when dealing with subreddits, as the user can sort and provider any listing options supported by `ListingOptions`. This also makes `SortedSubmissions()` obsolete.

Please note that I am a total Go beginner. Please advise on any code improvements and I will happily push new commits as needed.

jzelinskie commented on login_session.go:
```diff
@@ -109,9 +111,16 @@ func (s LoginSession) Clear() error {
+		return nil, err
+	}
+
+	baseUrl, err := url.Parse(fmt.Sprintf("http://www.reddit.com/%s/.json?%s", sort, v.Encode()))
```
There's no point in using `url.Parse` especially not if you aren't even checking the error.

jzelinskie commented on session.go:
```diff
@@ -28,81 +29,30 @@ func NewSession(useragent string) *Session {
-			}
-		}
+	if subreddit != "" {
+		/* if a subreddit was given, insert without memory garbage */
```
Only use `//` comments in Go

jzelinskie commented on session.go:
```diff
@@ -28,81 +29,30 @@ func NewSession(useragent string) *Session {
+		redditUrl = append(redditUrl, "")
+		copy(redditUrl[2:], redditUrl[1:])
+		redditUrl[1] = "/r"
 	}
```
IMO, this whole strategy of building a slice and then using `strings.Join` feels too hard to read.
It'd probably be better to be more verbose, simply to improve readability.

aggrolite commented:
Hi @jzelinskie, thank you for the feedback. I have pushed a new commit which should address your concerns here: https://github.com/aggrolite/geddit/commit/81e257b89ad1d4c1eccbcbd33c22ba296bb7ef84

jzelinskie commented on session.go:
```diff
@@ -28,81 +27,28 @@ func NewSession(useragent string) *Session {
-		return nil, err
+	// If subbreddit given, add to URL
+	if subreddit != "" {
+		baseUrl = fmt.Sprint(baseUrl + "/r/" + subreddit)
```
I don't think you need to use `fmt.Sprint` here at all.

jzelinskie commented:
looks good, if you remove the unnecessary `Sprint`, I'll merge it.

Thanks for the contributions @aggrolite!

aggrolite commented:
I have replaced the `Sprint()` call with the `+=` operator. A patch for README is included as well.

I've also added a `.gitignore`, which can later be expanded to hide more than just Vim swap files.

jzelinskie commented:
Can you rollback the `.gitignore` commit?
I feel like these are largely personal to the developer's workflow and shouldn't be forced on anyone who wants to work on the source.

aggrolite commented:
No problem, fixed.

jzelinskie commented:
The example should use `reddit.DefaultPopularity` instead of empty string.

aggrolite commented:
Ah, neat.

Do you think replacing the string "new" with the `NewSubmissions` constant should be done as well?

jzelinskie commented:
I actually edited my comment to include that before you posted ;)

aggrolite commented:
:+1: 

see how that looks

jzelinskie commented on README.md:
```diff
@@ -27,11 +27,22 @@ func main() {
+	}
+
+	// Used to specify default subreddit sort
+	defaultPop := geddit.DefaultPopularity
```
idk if this block of code is necessary, let's just inline `reddit.DefaultPopularity`.

aggrolite replied:
fixed here: e58266e1f07ae7cad0303f4ebd2a4d2ac3679495

jzelinskie commented:
lgtm, thanks for the contribution!
